Connect to backend

majacannavo opened the pull request:
Adds the following functionality:
- when a new playlist is created, its id is stored in state as editingPlaylistId, and its id is sent to a new table in the database called CurrentPlaylist, which just stores a single integer, the current playlist id.
- when new songs are added to the playlist being edited, they are sent to the database, and when they are deleted, they are deleted from the database 
- when the playlist being edited is deleted, it is deleted from the database
- the songs in the playlist currently being edited are stored in state as editingPlaylistSongs, and these are the ones that are displayed on the DJ page (NOT the songs from the current playlist, which we want to display on the listener page)

Commit:
current playlist stored in database

Commit:
passes linter checks

Commit:
changed name of state that stores editing playlist songs

brendanphilbin commented:
Looks good Maja, sorry I wasn't able to work on this today but I had to finally get Assignment 4 wrapped up. I'll spend tomorrow working on this as well as completing the test suite for the PlaylistExplorer

mylesstokowski commented on src/components/Playlist.js:
```diff
@@ -37,5 +37,8 @@ export default function Playlist({songs, deleteSong, addSong, mode}) {
 Playlist.propTypes = {
   songs:PropTypes.arrayOf(PropTypes.object).isRequired,
+  deleteSong:PropTypes.func,
+  addSong:PropTypes.func,
```
I like that you've renamed addSong to addSongToPlaylist elsewhere - I think that is more descriptive. Is the idea here that it's not renamed addSongToPlaylist because we're in the Playlist component so that's sort of implicit?

majacannavo replied:
Yeah that was kind of my train of thought but I can definitely change it here too for the sake of consistency if you think it'd be a good idea!

majacannavo replied:
And for the error about the unique time window, yes that's because we already have the current show's playlist at that time. If you want to try adding a new playlist you can change the current show and/or date

mylesstokowski commented on src/lib/playlist-table-utils.js:
```diff
@@ -67,7 +67,7 @@ export async function createPlaylist(input_show_id) {
     // THE +1 AFTER TODAY.GETDATE() SHOULD BE DELETED!
     // it was added for testing purposes so we can add more playlists
-    const dd = String(today.getDate() +1).padStart(2, "0");
+    const dd = String(today.getDate() -11).padStart(2, "0");
```
was the -11 also for similar testing purposes?

majacannavo replied:
Haha yes--I'm sorry I forgot to comment on that!

mylesstokowski reviewed (commented):
Awesome work, thanks!

I tried pulling it and got an error on clicking the new playlist button. I'm wondering if it's just happening because of the current data in the database that was added for testing – specifically that there's already a playlist in the database during the current time slot. If that's the case then I think we should be good to go. Here's the error message:

```
[Error: insert into `Playlist` (`show_id`, `time_window`) values (4, 'D05052021T09:00/10:00') - SQLITE_CONSTRAINT: UNIQUE constraint failed: Playlist.time_window] {
  errno: 19,
  code: 'SQLITE_CONSTRAINT'
}
```

mylesstokowski commented:
Linked this PR to two issues - pretty sure these are the correct ones.

mylesstokowski commented:
@majacannavo thanks for responding to all my comments - hopefully not too annoying! Looks great to me!

SimonScharf commented:
Everything looks great to me! Love the tests! the only reason I personally haven't merged it, is that when I run it and try to add a playlist I get an error response from the server, but that's probably an error on my end that we can figure out at our next meeting (probably in class Tuesday)

mylesstokowski commented:
Looks like there are a couple conflicting files – maybe good to merge this after resolving those?

brendanphilbin commented:
Yeah I think that the database conflict shouldnt be a problem, should be able to just choose one and go with it because once we migrate/seed it should be fine. Once this is merged I'm going to branch off and look to establish the CurrentPlaylist / MostRecentPlaylist component now that Maja set up the backend for that.

brendanphilbin commented:
And according to Andrews we shouldnt even have the database file in our repo so we could honestly just delete the file and that way everyone just holds on to a local copy I guess. Not sure if a database file is necessary for Heroku deployment

mylesstokowski commented:
> Once this is merged I'm going to branch off and look to establish the CurrentPlaylist / MostRecentPlaylist component now that Maja set up the backend for that.

Glad you said this here... in fact Assadou and I have been working on that component in the refreshListenerPage branch, as part of getting the current playlist to update at some time interval

> we shouldnt even have the database file in our repo

I was wondering about this when it was also causing problems for Assadou's PR

brendanphilbin commented:
Oh great to hear, in that case I'm going to try and get the PlaylistExplorer looking nicer and ready for a PR. I could also jump onto Maja's branch and help her with any outstanding issues on the DJ page

majacannavo commented:
Sorry I haven't been on top of responding to this conversation earlier--I can try to remove the database file from the commit and also fix the issues with index.js so that we can get this merged

Commit:
Merge branch 'main' into ConnectToBackend

Commit:
deleted SecureItems.js from index.js

Commit:
add dropdown menu to select current show

majacannavo commented:
I just added a dropdown menu (pulling from the PlaylistExplorer--thanks guys!) to select the current show (the one we are editing a playlist for)--which is stored in state. I don't think there's any reason to store it in the database since the playlist is just being edited by a single client. I also added a link to the DJ page without a login so that I could view it despite having problems logging in. This can definitely be deleted once we're all able to log in ok.
